### PR TITLE
fix: Showing STR error in case of DEVELOPMENT app config

### LINKED ISSUE #4903: Show debug error information on exception for development instance
For development instance of application, especially on obscure errors like 500, debug stack trace information should be shown to the user in the response JSON

This will severely cut down the reliance of client developers on server developers and enable them to create issues with proper data

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #4903 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by kushthedude** [comment]
@shreyanshdwivedi Found a quick fix so pushed the PR ,
@iamareebjamal Please check if it is acceptable

File: app/__init__.py [hunk 1]
```diff
@@ -245,7 +245,7 @@ def update_sent_state(sender=None, headers=None, **kwargs):
 
 @app.errorhandler(500)
 def internal_server_error(error):
-    if current_app.config['PROPOGATE_ERROR'] is True:
+    if current_app.config['DEVELOPMENT'] is True:
```
Comment (CosmicCoder96) [inline app/__init__.py]: This is redundant, propagate error is already true for development config.

https://github.com/fossasia/open-event-server/blob/8f2de81dbbfe179770f160f59fc365316ffed660/config.py#L119

**Comment by iamareebjamal** [comment]
Secondly, also do logger.exception so that the exception is at least logged on the console

**Comment by iamareebjamal** [comment]
Now check if it is working

**Comment by iamareebjamal** [comment]
@mrsaicharan1 @shreyanshdwivedi Can you please check?

File: app/__init__.py [hunk 2]
```diff
@@ -246,8 +247,9 @@ def update_sent_state(sender=None, headers=None, **kwargs):
 @app.errorhandler(500)
 def internal_server_error(error):
     if current_app.config['PROPOGATE_ERROR'] is True:
-        exc = JsonApiException({'pointer': ''}, str(error))
+        raise error
```
Comment (bhaveshAn) [inline app/__init__.py]: You have to display appropriate exception at API level and console level too.
In API as Json response and at console using `logger.exception`.

Comment (iamareebjamal) [reply app/__init__.py]: Nope, JSON will not be returned in case PROPAGATE_ERROR is True

**Comment by kushthedude** [comment]
> Even though better than 'Unknown error', str(error) is not enough. We want full stack strace of the error

Sorry @iamareebjamal I was AFK, That's why could not reply, Regarding the Previous change which i made did show the JSON error, But full stack trace was not visible

**Comment by iamareebjamal** [comment]
I have made certain changes. Please check

**Comment by kushthedude** [comment]
> @kushthedude Can you replicate an error and upload a screenshot of the stacktrace being shown?

@mrsaicharan1 I didn't try to check with the latest change, As the `ENV` was set to `production` in deployments, Which had been fixed by @iamareebjamal.

**Pull request closed without merging**